File: python/2.GreatestCommonDivisorOfStrings.py
def greatestCommonDivisorOfStrings():
    str1 = "ABABAB"
    str2 = "ABAB"

    a = max(len(str1), len(str2))
    b = len(str2) if a != len(str2) else len(str1)

    while b != 0:
        temp = b
        b = a % b
        a = temp

    return str1[:a] if str1 + str2 == str2 + str1 else ""  

print(greatestCommonDivisorOfStrings())

File: javascript/2.GreatestCommonDivisorOfStrings.js
function greatestCommonDivisorOfStrings(str1, str2) {
    let a = Math.max(str1.length, str2.length)
    let b = str1.length != a ? str1.lenght : str2.length

    while (b != 0) {
        let temp = b
        b = a % b
        a = temp
    }

    return str1 + str2 == str2 + str1 ? str1.substring(0, a) : ""
}


console.log("1. " + greatestCommonDivisorOfStrings("ABCABC", "ABC"))
console.log("2. " + greatestCommonDivisorOfStrings("ABABAB", "ABAB"))
console.log("3. " + greatestCommonDivisorOfStrings("XYZXYZXYZ", "XYZ"))
console.log("4. " + greatestCommonDivisorOfStrings("LEET", "CODE"))
console.log("5. " + greatestCommonDivisorOfStrings("ABCABCABC", "ABCAAA"))
